crates/core/src/db/datastore/system_tables.rs

```diff
@@ -235,231 +248,223 @@ st_fields_enum!(enum StVarFields {
+            col_list![StColumnFields::TableId.col_id(), StColumnFields::ColPos.col_id()],
+            None,
+        );
+
+    // System Table [ST_INDEX_NAME]
+    //
+    // | index_id | table_id | index_name  | index_algorithm            |
+    // |----------|----------|-------------|----------------------------|
+    // | 1        |          | "ix_sample" | btree({"columns": [1, 2]}) |
+    let st_index_type = builder.add_type::<StIndexRow>();
+    builder
+        .build_table(ST_INDEX_NAME, *st_index_type.as_ref().expect("should be ref"))
+        .with_type(TableType::System)
+        .with_auto_inc_primary_key(StIndexFields::IndexId);
+    // TODO(1.0): unique constraint on name?
+
+    // System Table [ST_SEQUENCE_NAME]
+    //
+    // | sequence_id | sequence_name     | increment | start | min_value | max_value | table_id | col_pos| allocated |
+    // |-------------|-------------------|-----------|-------|-----------|-----------|----------|--------|-----------|
+    // | 1           | "seq_customer_id" | 1         | 100   | 10        | 1200      | 1        | 1      | 200       |
+
+    let st_sequence_type = builder.add_type::<StSequenceRow>();
+    builder
+        .build_table(ST_SEQUENCE_NAME, *st_sequence_type.as_ref().expect("should be ref"))
+        .with_type(TableType::System)
+        .with_auto_inc_primary_key(StSequenceFields::SequenceId);
+    // TODO(1.0): unique constraint on name?
+
+    // System Table [ST_CONSTRAINT_NAME]
+    //
+    // | constraint_id | constraint_name      | table_id    | constraint_data    -------------|
+    // |---------------|-------------------- -|-------------|---------------------------------|
+    // | 1             | "unique_customer_id" | 1           | unique({"columns": [1, 2]})     |
+    let st_constraint_type = builder.add_type::<StConstraintRow>();
+    builder
+        .build_table(ST_CONSTRAINT_NAME, *st_constraint_type.as_ref().expect("should be ref"))
+        .with_type(TableType::System)
+        .with_auto_inc_primary_key(StConstraintFields::ConstraintId);
+    // TODO(1.0): unique constraint on name?
+
+    // System table [ST_MODULE_NAME]
+    // This table holds exactly one row, describing the latest version of the
+    // SpacetimeDB module associated with the database:
+    //
+    // * `database_address` is the [`Address`] of the database.
+    // * `owner_identity` is the [`Identity`] of the owner of the database.
+    // * `program_kind` is the [`ModuleKind`] (currently always [`WASM_MODULE`]).
+    // * `program_hash` is the [`Hash`] of the raw bytes of the (compiled) module.
+    // * `program_bytes` are the raw bytes of the (compiled) module.
+    //
+    // | database_address | owner_identity |  program_kind | program_bytes | program_hash        |
+    // |------------------|----------------|---------------|---------------|---------------------|
+    // | <bytes>          | <bytes>        |  0            | <bytes>       | <bytes>             |
```

Comment:
NIT: This diagram is missing `module_version`.

Reply:
Fixed, and I moved these to doc comments on the `St...Row` types since they make sense there.